MachineLearningPrototype/utils/preprocess_data.py

```diff
@@ -15,3 +15,24 @@ def extract_quotation_content(content):
     if (len(substring) >= 1):
         result=' '.join(substring)
         return result
+
+def extract_jenkins_link_and_testname(content):
+    pattern = r"((?:http|ftp|https):\/\/.+\/job\/Test_openjdk\d+.+\/\d+)" #For explanation visit https://regexr.com/60jmf
+    substring = re.findall(pattern, s)
+    result='\n'.join(substring)
+
+    pattern2 = r"(?<=\`).+?\/(?=\`)"
+    substring2 = re.findall(pattern2, s)
+    result2='\n'.join(substring2)
+    final_link = ''.join(('https://internal_jenkins/job/',result2))
+    
```

Comment:
We also need patterns to find test names, and save in a List too. One pattern can be "_FAILED", before it will be the test name, e.g. find `jdk_util_0_FAILED` and extract `jdk_util_0`; or find `jdk_math_0_FAILED` to extract `jdk_math_0`.